File: src/main/java/com/proiectPS/demo/DTO/TransactionRequest.java
package com.proiectPS.demo.DTO;


import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
public class TransactionRequest {

    public String userNickname;
    public String coinID;
    public double coinPrice;
    public double transactionAmount;
    public String currentDate;


    @Override
    public String toString() {
        return "TransactionRequest{ \n" +
                "userNickname='" + userNickname + '\'' + "\n" +
                ", coinID='" + coinID + '\'' +"\n" +
                ", coinPrice=" + coinPrice +"\n" +
                ", transactionAmount=" + transactionAmount +"\n" +
                ", currentDate='" + currentDate + '\'' +"\n" +
                '}' + "\n";
    }
}

File: frontEnd/TudorAngular/src/app/model/transaction-request.ts
export class TransactionRequest {

    public userNickname : string
    public coinID  :  string;
    public coinPrice : number;
    public transactionAmount : number;
    public currentDate : string;

    constructor(coinID : string, coinCurrentPrice : number, amount:  number, currentDate : string, userNickname : string ){
        this.coinID = coinID
        this.coinPrice = coinCurrentPrice
        this.transactionAmount = amount
        this.currentDate = currentDate
        this.userNickname = userNickname
    }

}
